File: follow-up-service/src/main/java/org/lorem/followupservice/domain/model/aggregates/Notification.java
package org.lorem.followupservice.domain.model.aggregates;

import jakarta.persistence.Entity;
import lombok.Getter;
import org.lorem.followupservice.domain.model.commands.CreateNotificationCommand;

@Getter
@Entity
public class Notification extends AuditableAbstractAggregateRoot<Notification> {

    private String title;

    private String description;

    private Long clientId;

    private Long consultationId;

    public Notification(CreateNotificationCommand command) {
        this.title = command.title();
        this.description = command.description();
        this.clientId = command.clientId();
        this.consultationId = command.legalCaseId();
    }

    public Notification() {

    }
}

File: LawConnectFrontEnd/src/app/followup/model/notification.ts
export class Notification {
  id: number;
  title: string;
  description: string;
  clientId: number;
  consultationId: number;

  constructor(id: number, title: string, description: string, clientId: number, consultationId: number) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.clientId = clientId;
    this.consultationId = consultationId;
  }
}
